R/backend-postgres.R

```diff
@@ -250,10 +250,24 @@ sql_query_explain.PqConnection <- function(con, sql, format = "text", ...) {
 #' @export
 sql_query_explain.PostgreSQL <- sql_query_explain.PqConnection
 
+#' Upsert using the `ON CONFLICT ... DO UPDATE` clause
+#'
+#' The `ON CONFLICT ... DO UPDATE` clause requires a unique index on the columns
+#' in `by`. One of the advantages is better concurrency behaviour. See the
+#' [Postgres Wiki](https://wiki.postgresql.org/wiki/UPSERT#SQL_MERGE_syntax) for
+#' more information on the advantages and disadvantages compared to other
+#' upsert implementations.
+#'
```

Comment:
Did you mean to add this documentation block to the generic?

Reply:
What's the best way to document the valid strings for `method` of the `sql_query_upsert()` methods?

Reply:
@hadley Do you have an idea

* how/where to document which methods are supported by which backend
* and where to describe the SQL code for a method

?

Reply:
It's probably easiest to document the methods with the generic (i.e. `@rdname sql_query_upsert`) since that way the majority of arguments are the same, but then you'll need to pull that out in to its own `.Rd` topic. Then I'd construct a hand written `@param method` that explained the basic options, with more details available in the `@details`. In the interests of getting this release out the door, I'd suggest focussing on the minimum possible for now, and we can come back to it after the release.